src/graph/graph.c

```diff
@@ -290,287 +124,291 @@ Graph *Graph_Get(RedisModuleCtx *ctx, RedisModuleString *graph_name) {
     return g;
 }
 
+// Return number of nodes graph can contain.
+size_t Graph_NodeCap(const Graph *g) {
+    return g->nodes->itemCap;
+}
+
 size_t Graph_NodeCount(const Graph *g) {
     assert(g);
     return g->nodes->itemCount;
 }
 
+// All graph matrices are required to be squared NXN
+// where N is Graph_RequiredMatrixDim.
+size_t Graph_RequiredMatrixDim(const Graph *g) {
+    // Matrix dimensions should be at least:
+    // Number of nodes + number of deleted nodes.
+    return g->nodes->itemCount + array_len(g->nodes->deletedIdx);
+}
+
 size_t Graph_EdgeCount(const Graph *g) {
     assert(g);
     return g->edges->itemCount;
 }
 
-void Graph_CreateNodes(Graph* g, size_t n, int* labels, DataBlockIterator **it) {
+int Graph_RelationTypeCount(const Graph *g) {
+    return array_len(g->_relations);
+}
+
+int Graph_LabelTypeCount(const Graph *g) {
+    return array_len(g->_labels);
+}
+
+void Graph_AllocateNodes(Graph* g, size_t n) {
+    assert(g && n >= 0);
```

Comment:
Second condition always true - consider changing `n` to `ssize_t` up through the function calls or dropping the condition.

Reply:
condition dropped.